### PR TITLE
Fix gaussian5x5 hexagon interleave

### PR DESCRIPTION
**Pull request by pranavb-ca**

### CODE REVIEW

**Comment by pranavb-ca** [comment]
@dsharletg : Please review.

File: src/CodeGen_Hexagon.cpp [hunk 1]
```diff
@@ -177,6 +177,7 @@ void CodeGen_Hexagon::init_module() {
     HvxIntrinsic intrinsic_wrappers[] = {
         // Zero/sign extension:
         { IPICK(Intrinsic::hexagon_V6_vzb), u16x2,  "zxt.vub", {u8x1} },
+        { IPICK(Intrinsic::hexagon_V6_vzb), i16x2,  "zxt.vub", {u8x1} },
```
Comment (dsharletg) [inline src/CodeGen_Hexagon.cpp]: I don't think this is actually needed, you just need the new patterns in HexagonOptimize.cpp. These two intrinsic declarations should generate the exact same LLVM functions.

Comment (pranavb-ca) [inline src/CodeGen_Hexagon.cpp]: Oh yes, you are right. Makes sense.

File: src/CodeGen_Hexagon.cpp [hunk 2]
```diff
@@ -702,9 +706,24 @@ void CodeGen_Hexagon::visit(const Mul *op) {
             std::swap(a, b);
 
         // Try to find an intrinsic for one of the operands being a scalar.
+        // All the non-widening vector by scalar multiplies expect a narrower
+        // scalar. Only two narrow types work, 8 bit and 16 bit. Start with
+        // the 8 bit ones
+        Expr narrow_b = lossless_cast(b.type().with_bits(8), b);
```
Comment (dsharletg) [inline src/CodeGen_Hexagon.cpp]: How about:

    for (int bits : {8, 16}) {
        Expr narrow_b = lossless_cast(b.type().with_bits(bits), b);
        Expr op_b = narrow_b.defined() ? narrow_b : b;
        value = call_intrin(op->type,
                             "halide.hexagon.mpyi" +
                             type_suffix(a, opb),
                             {a, opb},
                              true /*maybe*/);
        if (value) return;
    }

?

Comment (pranavb-ca) [inline src/CodeGen_Hexagon.cpp]: Ok, you got me. I just got lazy and didn't roll the loop. Will do

File: src/runtime/hvx_128.ll [hunk 3]
```diff
@@ -1,40 +1,40 @@
 declare void @llvm.trap() noreturn nounwind
 
-declare <32 x i32> @llvm.hexagon.V6.lo_128B(<64 x i32>)
-declare <32 x i32> @llvm.hexagon.V6.hi_128B(<64 x i32>)
-declare <64 x i32> @llvm.hexagon.V6.vshuffvdd_128B(<32 x i32>, <32 x i32>, i32)
+declare <32 x i32> @llvm.hexagon.V6.lo.128B(<64 x i32>)
```
Comment (dsharletg) [inline src/runtime/hvx_128.ll]: Thanks for fixing these. My bad, I should have tested this!

**Comment by dsharletg** [comment]
Thanks, this looks great. Merging.

**Pull request merged**